federatedscope/core/worker/client.py

```diff
@@ -445,15 +445,12 @@ def callback_funcs_for_finish(self, message: Message):
             self.trainer.update(message.content,
                                 strict=self._cfg.federate.share_local_model)
 
-        # TODO: more elegant here
         # Save final prediction result
         if self._cfg.data.type == 'cikmcup':
             # Evaluate
             self.trainer.evaluate(target_data_split_name='test')
-            # Save results
-            self.trainer.save_prediction(self._cfg.eval.prediction_path, self.ID, self._cfg.model.task)
-            logger.info(f"Client #{self.ID} finished saving prediction results in {os.path.abspath(self._cfg.eval.prediction_path)}")
-            logger.info(f"Please remember to save the prediction results elsewhere to avoid being covered by the new prediction results.")
+            self.trainer.save_prediction(self._cfg.outdir, self.ID, self._cfg.model.task)
+            logger.info(f"Client #{self.ID} finished saving prediction results in {os.path.abspath(self._cfg.outdir)}/predictions.csv")
```

Comment:
we'd better use `os.path.join` to concatenate them

Reply:
> we'd better use `os.path.join` to concatenate them

Modified accordingly